crypto/ake/enclave/src/lib.rs

```diff
@@ -645,77 +644,61 @@ impl<EI: EnclaveIdentity> AkeEnclaveState<EI> {
             }
         };
 
-        // Copy the public key into the report data.
+        let report_contents = EnclaveReportDataContents::new(
+            quote_nonce.clone(),
+            self.get_kex_identity(),
+            self.custom_identity.get_bytes_for_report(),
+        );
+
         let mut report_data = ReportData::default();
         let report_data_bytes: &mut [u8] = report_data.as_mut();
-        let identity = self.get_kex_identity();
-        let identity_bytes: &[u8] = identity.as_ref();
-        let custom_identity_bytes = self.custom_identity.get_bytes_for_report();
+        let identity_bytes: &[u8] = report_contents.key().as_ref();
         report_data_bytes[..identity_bytes.len()].copy_from_slice(identity_bytes);
-        report_data_bytes[identity_bytes.len()..].copy_from_slice(custom_identity_bytes.as_ref());
+        report_data_bytes[identity_bytes.len()..]
+            .copy_from_slice(report_contents.custom_identity());
```

Comment:
DCAP will be storing a hash of the `report_contents` in the `ReportData`, which means DCAP will need to provide the `report_contents` as part of it's version of the `VerificationReport`.

This EPID implementation was not changed to prevent breaking its `VerificationReport`.